File: linkedlistcycle.py
# Definition for singly-linked list.
from typing import Optional

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        x = []

        while (head and head.next != None):

            if (head in x):
                return True

            x.append(head)
            head = head.next;
        
        return False

File: linkedlistcycle.cpp
#include <vector>
#include <algorithm>

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
    bool hasCycle(ListNode *head) {
        
        ListNode* curr = head;
        std::vector<ListNode*> x {};
        while (curr != nullptr){
            auto it = std::find(x.begin(), x.end(), curr);
            if (it != x.end()){
                return true;
            }
            x.push_back(curr);
            curr = curr-> next;
        }
        return false;
    }
};
